server/src/main/java/org/opensearch/common/util/concurrent/SizeBlockingQueue.java

```diff
@@ -48,7 +48,7 @@
 public class SizeBlockingQueue<E> extends AbstractQueue<E> implements BlockingQueue<E> {
 
     private final BlockingQueue<E> queue;
-    private final int capacity;
+    protected volatile int capacity;
```

Comment:
I think this change should not be taken lightly: the truly fixed size queues (SizeBlockingQueue) would have to pay a price of accessing the volatile field which is only adjustable for resizeable queues.